source/adios2/toolkit/sst/dp/rdma_dp.c

```diff
@@ -617,6 +776,189 @@ static TimestepList GetStep(Rdma_WS_Stream Stream, long Timestep)
     return (Step);
 }
 
+#ifdef SST_HAVE_CRAY_CXI
+int get_cxi_auth_key_from_env(CP_Services Svcs, void *CP_Stream,
+                              struct _SstParams *Params,
+                              struct cxi_auth_key *key,
+                              char **used_device)
+{
+    int vni, first_vni, second_vni, svc_id;
+
+    // Just some safety against string processing.
+    size_t const no_infinite_loops = 10000;
+
+    // struct cxi_auth_key {
+    //     /* The CXI service assigned to the Domain and Endpoints. A CXI
+    //     service
+    //     * is associated with a set of local resource limits, VNIs, and
+    //     Traffic
+    //     * Classes.
+    //     *
+    //     * The svc_id used by an OFI Domain must match all Endpoints belonging
+    //     * to the Domain.
+    //     */
+    //     uint32_t svc_id;
+
+    //     /* The Virtual Network ID (VNI) assigned to the Endpoint. Two
+    //     Endpoints
+    //     * must use the same VNI in order to communicate.
+    //     *
+    //     * Note that while the CXI service may define one or more VNIs which a
+    //     * process can access, an Endpoint is assigned to only one.
+    //     */
+    //     uint16_t vni;
+    // };
```

Comment:
Might need to remove this comment due to licensing reasons? It's just a copy-pasted documentation from the CXI header.